nixos/modules/services/hardware/nvidia-container-toolkit-cdi-generator/cdi-generate.nix

```diff
@@ -0,0 +1,37 @@
+{ config, lib, pkgs }: let
+  mountOptions = { options = ["ro" "nosuid" "nodev" "bind"]; };
+  mounts = [
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-control";
+      containerPath = "/usr/bin/nvidia-cuda-mps-control"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-server";
+      containerPath = "/usr/bin/nvidia-cuda-mps-server"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-debugdump";
+      containerPath = "/usr/bin/nvidia-debugdump"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-powerd";
+      containerPath = "/usr/bin/nvidia-powerd"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-smi";
+      containerPath = "/usr/bin/nvidia-smi"; }
```

Comment:
By the way, I find it questionable that we expose those in the global `/usr/bin`. I'd say, we should by default limit ourselves to adding the respective `/nix/store` paths to `PATH`

Implementation detail, can be changed later. When writing the release notes I'd mention that `/usr/bin` isn't (yet) a part of the public interface of the nixos module and may get pruned in the future. The binaries being available through the `PATH` should be our promise, imo

Reply:
Although it's a good idea, it's not sadly not possible. This is because the OCI hooks cannot expand existing envvars: while it's possible to set, say `PATH` to `/something`, it's not possible to say `PATH=/somewthing:$PATH`, so we'd break the container.

I am mounting it on paths that are FHS friendly.

Reply:
That's maybe good enough. We can later introduce an option to control the PATH set by CDI, and we can populate it with the mounted binaries as well as the "well-known" (but empty) paths like `/usr/bin`